fvm/transaction_processor_test.go

```diff
@@ -0,0 +1,257 @@
+package fvm
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/sema"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func makeTwoAccounts(t *testing.T) (flow.Address, flow.Address, *state.State) {
+
+	ledger := state.NewMapLedger()
+	st := state.NewState(ledger)
+
+	a := flow.HexToAddress("1234")
+	b := flow.HexToAddress("5678")
+
+	//create accounts
+	accounts := state.NewAccounts(st)
+	err := accounts.Create(nil, a)
+	require.NoError(t, err)
+	err = accounts.Create(nil, b)
+	require.NoError(t, err)
+
+	err = st.Commit()
+	require.NoError(t, err)
+
+	return a, b, st
+}
+
+func TestAccountFreezing(t *testing.T) {
+
+	t.Run("setFrozenAccount can be enabled", func(t *testing.T) {
```

Comment:
Maybe add a test here that ensures that when account freezing is enabled, only the service account can call the function, and other locations (e.g. like here a transaction) can't call the function.

Reply:
Good idea, will do